keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlDeliveryConfigRepository.kt

```diff
@@ -105,13 +109,16 @@ class SqlDeliveryConfigRepository(
           .execute()
       }
       environments.forEach { environment ->
-        val environmentUID = jooq
-          .select(ENVIRONMENT.UID)
-          .from(ENVIRONMENT)
-          .where(ENVIRONMENT.DELIVERY_CONFIG_UID.eq(uid))
-          .and(ENVIRONMENT.NAME.eq(environment.name))
-          .fetchOne(ENVIRONMENT.UID)
-          ?: randomUID().toString().also {
+        val environmentUID = (
+          jooq
+            .select(ENVIRONMENT.UID)
+            .from(ENVIRONMENT)
+            .where(ENVIRONMENT.DELIVERY_CONFIG_UID.eq(uid))
+            .and(ENVIRONMENT.NAME.eq(environment.name))
+            .fetchOne(ENVIRONMENT.UID)
+            ?: randomUID().toString()
+          )
+          .also {
```

Comment:
Doesn’t this change mean we’ll attempt to insert duplicate environment ids?

Reply:
No, this is fixing a preexisting bug that meant updates to environment constraints and notifications would never be persisted.  Can't duplicate environment ids, it's a primary key, but we want the insert to always run to get the `onDuplicateKeyUpdate()`.